### PR TITLE
Fix and add functions in speaker utils

### PR DESCRIPTION
**Pull request by tango4j**
# What does this PR do ?

This PR fixes bugs for certain cases and adds few functions. Previously, write_rttm2manifest function had following issues:

(1) Offset is not considered
(2) Occasionally, the function was returning the segments outside the given (offset, offset+duration) range.

The newly written part is compared with the original code, and confirmed to 

**Collection**: [Note which collection this PR will affect]

ASR

# Changelog 
- Add specific line by line info of high level changes in this PR.

Added following functions in speaker_utils.py.
These functions are needed to extract oracle VAD stamp from manfiest file, considering the offset and duration.
In addition, these functions are designed for the future use for streaming diarization. 

get_uniq_id_with_dur()
getSubRangeList()
getMinMaxOfRangeList()
getMergedRanges()
int2fl()
fl2int()
combine_int_overlaps()
combine_float_overlaps()
getOverlapRange()
isOverlap()

and modified following functions:
write_rttm2manifest() - to resolve existing issues
segments_manifest_to_subsegments_manifest() - Now, if include_uniq_id = True, the function adds uniq_id to the created subsegment manifest file. This is for using segmentation files for truncated (chopped) diarization dataset, where uniq_id is overlapped if not specified. (One audio file creates multiple 30~40sec utterances and these short utterances should be identifed too.)

Details can be found in the docstring in each function definition.

# Before your PR is "Ready for review"
**Pre checks**:
- [x] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [x] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [x] New Feature
- [x] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/collections/asr/parts/utils/speaker_utils.py [hunk 1]
```diff
@@ -27,26 +30,46 @@
...
         return basename
     else:
         raise TypeError("input must be filepath string")
 
 
+def get_uniq_id_with_dur(meta, deci=3):
+    """
+    Return basename with offset and end time labels
+    """
+    bare_uniq_id = meta['audio_filepath'].split('/')[-1].split('.wav')[0]
```
Comment (nithinraok) [inline nemo/collections/asr/parts/utils/speaker_utils.py:53]: same here as mentioned above for uniqname fix

File: nemo/collections/asr/parts/utils/speaker_utils.py [hunk 2]
```diff
@@ -438,88 +465,326 @@ def score_labels(AUDIO_RTTM_MAP, all_reference, all_hypothesis, collar=0.25, ign
...
+    else:
+        sound = sf.SoundFile(audio_path)
+        duration = sound.frames / sound.samplerate
+        offset = 0.0
+    return offset, duration
+
+
+def write_overlap_segments(outfile, AUDIO_RTTM_MAP, uniq_id, overlap_range_list, include_uniq_id, deci=5):
+    """
+    Write the json dictionary into the specified file.
```
Comment (nithinraok) [inline nemo/collections/asr/parts/utils/speaker_utils.py:510]: specified file -> specified manifest file

**Review by nithinraok — CHANGES_REQUESTED** [review]

File: nemo/collections/asr/parts/utils/speaker_utils.py [hunk 3]
```diff
@@ -27,26 +30,46 @@
...
 """
 
 
 def get_uniqname_from_filepath(filepath):
-    "return base name from provided filepath"
+    """
+    Return base name from provided filepath
+    """
     if type(filepath) is str:
         basename = os.path.basename(filepath).rsplit('.', 1)[0]
```
[the remark this answers is not in the document]
Comment (tango4j) [reply nemo/collections/asr/parts/utils/speaker_utils.py:43]: Replaced with the line Yang has changed.

File: nemo/collections/asr/parts/utils/speaker_utils.py (hunk 2 above)
Comment (tango4j) [reply nemo/collections/asr/parts/utils/speaker_utils.py:510]: added "manifest'.

**Pull request merged**